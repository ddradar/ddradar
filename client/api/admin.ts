import type { NuxtHTTPInstance } from '@nuxt/http'

import type { SongInfo } from '~/api/song'
import type { NotificationSchema } from '~/core/db/notification'

type NotificationRequest = Omit<
  NotificationSchema,
  'id' | 'sender' | 'timeStamp'
> & {
  id?: string
  timeStamp?: number
}

/**
 * Call "Post Notification" API. (Admin only)
 * @see https://github.com/ddradar/ddradar/tree/master/api/postNotification
 */
export function postNotification(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  body: NotificationRequest
) {
  return $http.$post<NotificationRequest>('/api/v1/admin/notification', {
    ...body,
    sender: 'SYSTEM',
  })
}

/**
 * Call "Post Song Information" API. (Admin only)
 * @see https://github.com/ddradar/ddradar/tree/master/api/postSongInfo
 */
export function postSongInfo(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  body: SongInfo
) {
  return $http.$post<SongInfo>('/api/v1/admin/songs', body)
}
