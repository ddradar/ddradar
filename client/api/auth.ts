import type { Api } from '@ddradar/core'
import type { NuxtHTTPInstance } from '@nuxt/http'

type AuthResult = {
  clientPrincipal: Api.ClientPrincipal | null
}

/**
 * Call "Get Credentials" API.
 * @see https://docs.microsoft.com/azure/static-web-apps/user-information
 */
export async function getClientPrincipal(
  $http: Pick<NuxtHTTPInstance, '$get'>
) {
  const { clientPrincipal } = await $http.$get<AuthResult>('/.auth/me')
  return clientPrincipal
}
