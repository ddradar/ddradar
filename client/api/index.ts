export const apiPrefix = '/api/v1'
