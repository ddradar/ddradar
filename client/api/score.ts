import type { Api } from '@ddradar/core'
import type { NuxtHTTPInstance } from '@nuxt/http'

import { apiPrefix } from '~/api'

/**
 * Call "Get Chart Score" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/scores__id__style__difficulty--get/
 */
export function getChartScore(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  songId: string,
  playStyle: 1 | 2,
  difficulty: 0 | 1 | 2 | 3 | 4,
  scope?: 'private' | 'medium' | 'full'
) {
  const query = scope ? `?scope=${scope}` : ''
  return $http.$get<Api.ScoreInfo[]>(
    `${apiPrefix}/scores/${songId}/${playStyle}/${difficulty}${query}`
  )
}

/**
 * Call "Post Chart Score" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/scores__id__style__difficulty--post/
 */
export function postChartScore(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  songId: string,
  playStyle: 1 | 2,
  difficulty: 0 | 1 | 2 | 3 | 4,
  score: Api.ScoreBody
) {
  return $http.$post(
    `${apiPrefix}/scores/${songId}/${playStyle}/${difficulty}`,
    score
  )
}

/**
 * Call "Delete Chart Score" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/scores__id__style__difficulty--delete/
 */
export function deleteChartScore(
  $http: Pick<NuxtHTTPInstance, 'delete'>,
  songId: string,
  playStyle: 1 | 2,
  difficulty: 0 | 1 | 2 | 3 | 4
) {
  return $http.delete(`/api/v1/scores/${songId}/${playStyle}/${difficulty}`)
}

/**
 * Call "Post Song Scores" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/scores__id--post/
 */
export function postSongScores(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  songId: string,
  scores: Api.ScoreListBody[]
) {
  return $http.$post<Api.ScoreInfo[]>(`${apiPrefix}/scores/${songId}`, scores)
}
