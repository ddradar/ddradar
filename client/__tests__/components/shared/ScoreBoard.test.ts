import type { Api } from '@ddradar/core'
import { Song } from '@ddradar/core'
import {
  privateUser,
  publicUser,
  testSongData,
} from '@ddradar/core/__tests__/data'
import {
  createLocalVue,
  mount,
  RouterLinkStub,
  shallowMount,
} from '@vue/test-utils'
import Buefy from 'buefy'
import { mocked } from 'ts-jest/utils'
import VueI18n from 'vue-i18n'

import { getChartScore } from '~/api/score'
import ScoreBoard from '~/components/shared/ScoreBoard.vue'

jest.mock('~/api/score')
jest.mock('@ddradar/core')
const localVue = createLocalVue()
localVue.use(Buefy)
localVue.use(VueI18n)

describe('/components/shared/ScoreBoard.vue', () => {
  const info = { ...testSongData, charts: undefined }
  const chart = { ...testSongData.charts[0] }
  const propsData = { info, chart }
  const $accessor = { isLoggedIn: true, user: { ...privateUser } }
  const scores = [
    {
      userId: '0',
      userName: '0',
      clearLamp: 7,
      score: 1000000,
      exScore: chart.notes * 3,
      isArea: true,
    },
    {
      userId: '13',
      userName: '13',
      clearLamp: 6,
      score: 999950,
      exScore: chart.notes * 3 - 5,
      isArea: true,
    },
    {
      userId: '1user',
      userName: publicUser.name,
      clearLamp: 6,
      score: 999950,
      exScore: chart.notes * 3 - 5,
    },
    {
      userId: privateUser.id,
      userName: privateUser.name,
      clearLamp: 6,
      score: 999930,
      exScore: chart.notes * 3 - 5,
    },
  ]
  const i18n = new VueI18n({ locale: 'ja', silentFallbackWarn: true })
  const mocks = {
    $accessor,
    $http: {},
    $fetchState: { pending: false },
    $fetch: jest.fn(),
  }
  const options = { localVue, propsData, mocks, i18n }
  let wrapper: ReturnType<typeof shallowMount>
  beforeEach(() => {
    wrapper = shallowMount(ScoreBoard, options)
    mocked(Song.isDeletedOnGate).mockReturnValue(false)
  })

  describe.each(['ja', 'en'])('{ locale: %s } snapshot test', locale => {
    const i18n = new VueI18n({ locale, silentFallbackWarn: true })
    const stubs = { NuxtLink: RouterLinkStub, ScoreBadge: true }
    const mocks = {
      $accessor: { isLoggedIn: false },
      $fetchState: { pending: false },
    }

    test('{ loading: true } renders loading spin', async () => {
      // Arrange
      const mocks = {
        $accessor: { isLoggedIn: false },
        $fetchState: { pending: true },
      }
      const data = () => ({ scores: [] })
      const options = { localVue, propsData, mocks, stubs, i18n, data }

      // Act
      const wrapper = mount(ScoreBoard, options)
      await wrapper.vm.$nextTick()

      // Assert
      expect(wrapper).toMatchSnapshot()
    })
    test('{ loading: false, scores: [...] } renders score list', async () => {
      // Arrange
      const data = () => ({ scores })
      const options = { localVue, propsData, mocks, stubs, i18n, data }

      // Act
      const wrapper = mount(ScoreBoard, options)
      await wrapper.vm.$nextTick()

      // Assert
      expect(wrapper).toMatchSnapshot()
    })
    test('{ loading: false, scores: [] } renders empty', async () => {
      // Arrange
      const data = () => ({ scores: [] })
      const options = { localVue, propsData, mocks, stubs, i18n, data }

      // Act
      const wrapper = mount(ScoreBoard, options)
      await wrapper.vm.$nextTick()

      // Assert
      expect(wrapper).toMatchSnapshot()
    })
    test('renders edit button if loggedIn', async () => {
      // Arrange
      const mocks = { $accessor, $fetchState: { pending: false } }
      const options = { localVue, propsData, mocks, stubs, i18n }

      // Act
      const wrapper = mount(ScoreBoard, options)
      await wrapper.vm.$nextTick()

      // Assert
      expect(wrapper).toMatchSnapshot()
    })
    test('not renders import button if deleted song', async () => {
      // Arrange
      mocked(Song.isDeletedOnGate).mockReturnValueOnce(true)
      const mocks = { $accessor, $fetchState: { pending: false } }
      const options = { localVue, propsData, mocks, stubs, i18n }

      // Act
      const wrapper = mount(ScoreBoard, options)
      await wrapper.vm.$nextTick()

      // Assert
      expect(wrapper).toMatchSnapshot()
    })
  })

  // Lifecycle
  describe('fetch()', () => {
    mocked(getChartScore).mockResolvedValue(scores as Api.ScoreInfo[])
    beforeEach(() => mocked(getChartScore).mockClear())

    test('{ fetchAllData: false } calls getChartScore(medium)', async () => {
      // Arrange - Act
      // @ts-ignore
      await wrapper.vm.$options.fetch?.call(wrapper.vm)

      // Assert
      expect(mocked(getChartScore)).toBeCalledWith(
        mocks.$http,
        info.id,
        chart.playStyle,
        chart.difficulty,
        'medium'
      )
    })
    test('{ fetchAllData: true } calls getChartScore(full)', async () => {
      // Arrange - Act
      wrapper.setData({ fetchAllData: true })
      // @ts-ignore
      await wrapper.vm.$options.fetch?.call(wrapper.vm)

      // Assert
      expect(mocked(getChartScore)).toBeCalledWith(
        mocks.$http,
        info.id,
        chart.playStyle,
        chart.difficulty,
        'full'
      )
    })
    test.each(['ja', 'en'])('sets { locale: %s } areaName', async locale => {
      // Arrange
      const i18n = new VueI18n({ locale, silentFallbackWarn: true })
      const mocks = { $accessor, $http: {}, $fetchState: { pending: true } }
      const options = { localVue, propsData, mocks, i18n }
      const wrapper = shallowMount(ScoreBoard, options)

      // Act
      // @ts-ignore
      await wrapper.vm.$options.fetch?.call(wrapper.vm)

      // Assert
      expect(wrapper.vm.$data.scores).toMatchSnapshot()
    })
  })

  // Method
  describe('launchScoreEditor()', () => {
    const mocks = {
      $accessor,
      $buefy: { modal: { open: jest.fn(_ => ({ $on: jest.fn() })) } },
      $fetchState: { pending: false },
    }
    const options = { localVue, propsData, mocks, i18n }
    const wrapper = shallowMount(ScoreBoard, options)
    beforeEach(() => mocks.$buefy.modal.open.mockClear())

    test('calls $buefy.modal.open', () => {
      // Arrange - Act
      // @ts-ignore
      wrapper.vm.launchScoreEditor()

      // Assert
      expect(mocks.$buefy.modal.open).toBeCalled()
    })
  })
  describe('launchScoreImporter()', () => {
    const mocks = {
      $accessor,
      $buefy: { modal: { open: jest.fn(_ => ({ $on: jest.fn() })) } },
      $fetchState: { pending: false },
    }
    const options = { localVue, propsData, mocks, i18n }
    const wrapper = shallowMount(ScoreBoard, options)
    beforeEach(() => mocks.$buefy.modal.open.mockClear())

    test('calls $buefy.modal.open', () => {
      // Arrange - Act
      // @ts-ignore
      wrapper.vm.launchScoreImporter()

      // Assert
      expect(mocks.$buefy.modal.open).toBeCalled()
    })
  })
  describe('fetchAllScores()', () => {
    mocked(getChartScore).mockResolvedValue(scores as Api.ScoreInfo[])
    beforeEach(() => mocks.$fetch.mockClear())

    test('sets { fetchAllData = true } and calls $fetch()', () => {
      // Arrange - Act
      // @ts-ignore
      wrapper.vm.fetchAllScores()

      // Assert
      expect(wrapper.vm.$data.fetchAllData).toBe(true)
      expect(mocks.$fetch).toBeCalled()
    })
  })
})
