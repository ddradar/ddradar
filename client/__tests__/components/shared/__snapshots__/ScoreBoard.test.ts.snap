// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/shared/ScoreBoard.vue { locale: en } snapshot test { loading: false, scores: [...] } renders score list 1`] = `
<div class="collapse card">
  <div class="collapse-trigger">
    <header role="button" aria-controls="SP-BEGINNER (4)" class="card-header is-beginner">
      <div class="card-header-title">SP-BEGINNER (4)</div> <a class="card-header-icon"><span class="icon"><i class="mdi mdi-menu-up mdi-24px"></i></span></a>
    </header>
  </div>
  <transition-stub name="slide">
    <div id="SP-BEGINNER (4)" class="collapse-content" style="display: none;">
      <div class="card-content">
        <div class="table-container">
          <div class="b-table">
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <div class="table-wrapper">
              <table class="table is-striped is-narrow">
                <thead>
                  <tr>
                    <!---->
                    <!---->
                    <th class="">
                      <div class="th-wrap"><span class="is-relative"> Name <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-centered"><span class="is-relative"> Score <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-numeric"><span class="is-relative"> EX <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <!---->
                  </tr>
                  <!---->
                  <!---->
                </thead>
                <tbody>
                  <tr draggable="false" class="">
                    <!---->
                    <!---->
                    <td data-label="Name" class=""><span class="is-size-7">0</span></td>
                    <td data-label="Score" class="has-text-centered">
                      <scorebadge-stub lamp="7" score="1000000"></scorebadge-stub>
                    </td>
                    <td data-label="EX" class="has-text-right"><span class="is-size-7">414</span></td>
                    <!---->
                  </tr>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <!---->
                  <tr draggable="false" class="">
                    <!---->
                    <!---->
                    <td data-label="Name" class=""><span class="is-size-7">13</span></td>
                    <td data-label="Score" class="has-text-centered">
                      <scorebadge-stub lamp="6" score="999950"></scorebadge-stub>
                    </td>
                    <td data-label="EX" class="has-text-right"><span class="is-size-7">409</span></td>
                    <!---->
                  </tr>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <!---->
                  <tr draggable="false" class="">
                    <!---->
                    <!---->
                    <td data-label="Name" class=""><a class="is-size-7">
                        Public User
                      </a></td>
                    <td data-label="Score" class="has-text-centered">
                      <scorebadge-stub lamp="6" score="999950"></scorebadge-stub>
                    </td>
                    <td data-label="EX" class="has-text-right"><span class="is-size-7">409</span></td>
                    <!---->
                  </tr>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <!---->
                  <tr draggable="false" class="">
                    <!---->
                    <!---->
                    <td data-label="Name" class=""><a class="is-size-7">
                        Private User
                      </a></td>
                    <td data-label="Score" class="has-text-centered">
                      <scorebadge-stub lamp="6" score="999930"></scorebadge-stub>
                    </td>
                    <td data-label="EX" class="has-text-right"><span class="is-size-7">409</span></td>
                    <!---->
                  </tr>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <!---->
                  <!---->
                </tbody>
                <!---->
              </table>
              <!---->
            </div>
            <!---->
          </div>
        </div>
      </div>
      <footer class="card-footer">
        <!---->
        <!----> <a class="card-footer-item">
          Show All
        </a>
      </footer>
    </div>
  </transition-stub>
</div>
`;

exports[`/components/shared/ScoreBoard.vue { locale: en } snapshot test { loading: false, scores: [] } renders empty 1`] = `
<div class="collapse card">
  <div class="collapse-trigger">
    <header role="button" aria-controls="SP-BEGINNER (4)" class="card-header is-beginner">
      <div class="card-header-title">SP-BEGINNER (4)</div> <a class="card-header-icon"><span class="icon"><i class="mdi mdi-menu-up mdi-24px"></i></span></a>
    </header>
  </div>
  <transition-stub name="slide">
    <div id="SP-BEGINNER (4)" class="collapse-content" style="display: none;">
      <div class="card-content">
        <div class="table-container">
          <div class="b-table">
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <div class="table-wrapper">
              <table class="table is-striped is-narrow">
                <thead>
                  <tr>
                    <!---->
                    <!---->
                    <th class="">
                      <div class="th-wrap"><span class="is-relative"> Name <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-centered"><span class="is-relative"> Score <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-numeric"><span class="is-relative"> EX <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <!---->
                  </tr>
                  <!---->
                  <!---->
                </thead>
                <tbody>
                  <tr class="is-empty">
                    <td colspan="3">
                      <section class="section">
                        <div class="content has-text-grey has-text-centered">
                          <p>No Data</p>
                        </div>
                      </section>
                    </td>
                  </tr>
                </tbody>
                <!---->
              </table>
              <!---->
            </div>
            <!---->
          </div>
        </div>
      </div>
      <footer class="card-footer">
        <!---->
        <!----> <a class="card-footer-item">
          Show All
        </a>
      </footer>
    </div>
  </transition-stub>
</div>
`;

exports[`/components/shared/ScoreBoard.vue { locale: en } snapshot test { loading: true } renders loading spin 1`] = `
<div class="collapse card">
  <div class="collapse-trigger">
    <header role="button" aria-controls="SP-BEGINNER (4)" class="card-header is-beginner">
      <div class="card-header-title">SP-BEGINNER (4)</div> <a class="card-header-icon"><span class="icon"><i class="mdi mdi-menu-up mdi-24px"></i></span></a>
    </header>
  </div>
  <transition-stub name="slide">
    <div id="SP-BEGINNER (4)" class="collapse-content" style="display: none;">
      <div class="card-content">
        <div class="table-container">
          <div class="b-table">
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <div class="table-wrapper">
              <table class="table is-striped is-narrow">
                <thead>
                  <tr>
                    <!---->
                    <!---->
                    <th class="">
                      <div class="th-wrap"><span class="is-relative"> Name <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-centered"><span class="is-relative"> Score <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-numeric"><span class="is-relative"> EX <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <!---->
                  </tr>
                  <!---->
                  <!---->
                </thead>
                <tbody>
                  <tr class="is-empty">
                    <td colspan="3">
                      <section class="section">
                        <div class="content has-text-grey has-text-centered">
                          <p>No Data</p>
                        </div>
                      </section>
                    </td>
                  </tr>
                </tbody>
                <!---->
              </table>
              <transition-stub name="fade">
                <div class="loading-overlay is-active">
                  <div class="loading-background"></div>
                  <div class="loading-icon"></div>
                </div>
              </transition-stub>
            </div>
            <!---->
          </div>
        </div>
      </div>
      <footer class="card-footer">
        <!---->
        <!----> <a class="card-footer-item">
          Show All
        </a>
      </footer>
    </div>
  </transition-stub>
</div>
`;

exports[`/components/shared/ScoreBoard.vue { locale: en } snapshot test renders edit button if loggedIn 1`] = `
<div class="collapse card">
  <div class="collapse-trigger">
    <header role="button" aria-controls="SP-BEGINNER (4)" class="card-header is-beginner">
      <div class="card-header-title">SP-BEGINNER (4)</div> <a class="card-header-icon"><span class="icon"><i class="mdi mdi-menu-up mdi-24px"></i></span></a>
    </header>
  </div>
  <transition-stub name="slide">
    <div id="SP-BEGINNER (4)" class="collapse-content" style="display: none;">
      <div class="card-content">
        <div class="table-container">
          <div class="b-table">
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <div class="table-wrapper">
              <table class="table is-striped is-narrow">
                <thead>
                  <tr>
                    <!---->
                    <!---->
                    <th class="">
                      <div class="th-wrap"><span class="is-relative"> Name <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-centered"><span class="is-relative"> Score <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-numeric"><span class="is-relative"> EX <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <!---->
                  </tr>
                  <!---->
                  <!---->
                </thead>
                <tbody>
                  <tr class="is-empty">
                    <td colspan="3">
                      <section class="section">
                        <div class="content has-text-grey has-text-centered">
                          <p>No Data</p>
                        </div>
                      </section>
                    </td>
                  </tr>
                </tbody>
                <!---->
              </table>
              <!---->
            </div>
            <!---->
          </div>
        </div>
      </div>
      <footer class="card-footer"><a class="card-footer-item">
          Edit
        </a> <a class="card-footer-item">
          Import
        </a> <a class="card-footer-item">
          Show All
        </a></footer>
    </div>
  </transition-stub>
</div>
`;

exports[`/components/shared/ScoreBoard.vue { locale: ja } snapshot test { loading: false, scores: [...] } renders score list 1`] = `
<div class="collapse card">
  <div class="collapse-trigger">
    <header role="button" aria-controls="SP-BEGINNER (4)" class="card-header is-beginner">
      <div class="card-header-title">SP-BEGINNER (4)</div> <a class="card-header-icon"><span class="icon"><i class="mdi mdi-menu-up mdi-24px"></i></span></a>
    </header>
  </div>
  <transition-stub name="slide">
    <div id="SP-BEGINNER (4)" class="collapse-content" style="display: none;">
      <div class="card-content">
        <div class="table-container">
          <div class="b-table">
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <div class="table-wrapper">
              <table class="table is-striped is-narrow">
                <thead>
                  <tr>
                    <!---->
                    <!---->
                    <th class="">
                      <div class="th-wrap"><span class="is-relative"> ユーザー名 <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-centered"><span class="is-relative"> スコア <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-numeric"><span class="is-relative"> EX <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <!---->
                  </tr>
                  <!---->
                  <!---->
                </thead>
                <tbody>
                  <tr draggable="false" class="">
                    <!---->
                    <!---->
                    <td data-label="ユーザー名" class=""><span class="is-size-7">0</span></td>
                    <td data-label="スコア" class="has-text-centered">
                      <scorebadge-stub lamp="7" score="1000000"></scorebadge-stub>
                    </td>
                    <td data-label="EX" class="has-text-right"><span class="is-size-7">414</span></td>
                    <!---->
                  </tr>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <!---->
                  <tr draggable="false" class="">
                    <!---->
                    <!---->
                    <td data-label="ユーザー名" class=""><span class="is-size-7">13</span></td>
                    <td data-label="スコア" class="has-text-centered">
                      <scorebadge-stub lamp="6" score="999950"></scorebadge-stub>
                    </td>
                    <td data-label="EX" class="has-text-right"><span class="is-size-7">409</span></td>
                    <!---->
                  </tr>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <!---->
                  <tr draggable="false" class="">
                    <!---->
                    <!---->
                    <td data-label="ユーザー名" class=""><a class="is-size-7">
                        Public User
                      </a></td>
                    <td data-label="スコア" class="has-text-centered">
                      <scorebadge-stub lamp="6" score="999950"></scorebadge-stub>
                    </td>
                    <td data-label="EX" class="has-text-right"><span class="is-size-7">409</span></td>
                    <!---->
                  </tr>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <!---->
                  <tr draggable="false" class="">
                    <!---->
                    <!---->
                    <td data-label="ユーザー名" class=""><a class="is-size-7">
                        Private User
                      </a></td>
                    <td data-label="スコア" class="has-text-centered">
                      <scorebadge-stub lamp="6" score="999930"></scorebadge-stub>
                    </td>
                    <td data-label="EX" class="has-text-right"><span class="is-size-7">409</span></td>
                    <!---->
                  </tr>
                  <transition-stub name="">
                    <!---->
                  </transition-stub>
                  <!---->
                  <!---->
                </tbody>
                <!---->
              </table>
              <!---->
            </div>
            <!---->
          </div>
        </div>
      </div>
      <footer class="card-footer">
        <!---->
        <!----> <a class="card-footer-item">
          全件表示
        </a>
      </footer>
    </div>
  </transition-stub>
</div>
`;

exports[`/components/shared/ScoreBoard.vue { locale: ja } snapshot test { loading: false, scores: [] } renders empty 1`] = `
<div class="collapse card">
  <div class="collapse-trigger">
    <header role="button" aria-controls="SP-BEGINNER (4)" class="card-header is-beginner">
      <div class="card-header-title">SP-BEGINNER (4)</div> <a class="card-header-icon"><span class="icon"><i class="mdi mdi-menu-up mdi-24px"></i></span></a>
    </header>
  </div>
  <transition-stub name="slide">
    <div id="SP-BEGINNER (4)" class="collapse-content" style="display: none;">
      <div class="card-content">
        <div class="table-container">
          <div class="b-table">
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <div class="table-wrapper">
              <table class="table is-striped is-narrow">
                <thead>
                  <tr>
                    <!---->
                    <!---->
                    <th class="">
                      <div class="th-wrap"><span class="is-relative"> ユーザー名 <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-centered"><span class="is-relative"> スコア <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-numeric"><span class="is-relative"> EX <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <!---->
                  </tr>
                  <!---->
                  <!---->
                </thead>
                <tbody>
                  <tr class="is-empty">
                    <td colspan="3">
                      <section class="section">
                        <div class="content has-text-grey has-text-centered">
                          <p>データがありません</p>
                        </div>
                      </section>
                    </td>
                  </tr>
                </tbody>
                <!---->
              </table>
              <!---->
            </div>
            <!---->
          </div>
        </div>
      </div>
      <footer class="card-footer">
        <!---->
        <!----> <a class="card-footer-item">
          全件表示
        </a>
      </footer>
    </div>
  </transition-stub>
</div>
`;

exports[`/components/shared/ScoreBoard.vue { locale: ja } snapshot test { loading: true } renders loading spin 1`] = `
<div class="collapse card">
  <div class="collapse-trigger">
    <header role="button" aria-controls="SP-BEGINNER (4)" class="card-header is-beginner">
      <div class="card-header-title">SP-BEGINNER (4)</div> <a class="card-header-icon"><span class="icon"><i class="mdi mdi-menu-up mdi-24px"></i></span></a>
    </header>
  </div>
  <transition-stub name="slide">
    <div id="SP-BEGINNER (4)" class="collapse-content" style="display: none;">
      <div class="card-content">
        <div class="table-container">
          <div class="b-table">
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <div class="table-wrapper">
              <table class="table is-striped is-narrow">
                <thead>
                  <tr>
                    <!---->
                    <!---->
                    <th class="">
                      <div class="th-wrap"><span class="is-relative"> ユーザー名 <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-centered"><span class="is-relative"> スコア <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-numeric"><span class="is-relative"> EX <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <!---->
                  </tr>
                  <!---->
                  <!---->
                </thead>
                <tbody>
                  <tr class="is-empty">
                    <td colspan="3">
                      <section class="section">
                        <div class="content has-text-grey has-text-centered">
                          <p>データがありません</p>
                        </div>
                      </section>
                    </td>
                  </tr>
                </tbody>
                <!---->
              </table>
              <transition-stub name="fade">
                <div class="loading-overlay is-active">
                  <div class="loading-background"></div>
                  <div class="loading-icon"></div>
                </div>
              </transition-stub>
            </div>
            <!---->
          </div>
        </div>
      </div>
      <footer class="card-footer">
        <!---->
        <!----> <a class="card-footer-item">
          全件表示
        </a>
      </footer>
    </div>
  </transition-stub>
</div>
`;

exports[`/components/shared/ScoreBoard.vue { locale: ja } snapshot test renders edit button if loggedIn 1`] = `
<div class="collapse card">
  <div class="collapse-trigger">
    <header role="button" aria-controls="SP-BEGINNER (4)" class="card-header is-beginner">
      <div class="card-header-title">SP-BEGINNER (4)</div> <a class="card-header-icon"><span class="icon"><i class="mdi mdi-menu-up mdi-24px"></i></span></a>
    </header>
  </div>
  <transition-stub name="slide">
    <div id="SP-BEGINNER (4)" class="collapse-content" style="display: none;">
      <div class="card-content">
        <div class="table-container">
          <div class="b-table">
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <div class="table-wrapper">
              <table class="table is-striped is-narrow">
                <thead>
                  <tr>
                    <!---->
                    <!---->
                    <th class="">
                      <div class="th-wrap"><span class="is-relative"> ユーザー名 <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-centered"><span class="is-relative"> スコア <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <th class="">
                      <div class="th-wrap is-numeric"><span class="is-relative"> EX <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
                    </th>
                    <!---->
                  </tr>
                  <!---->
                  <!---->
                </thead>
                <tbody>
                  <tr class="is-empty">
                    <td colspan="3">
                      <section class="section">
                        <div class="content has-text-grey has-text-centered">
                          <p>データがありません</p>
                        </div>
                      </section>
                    </td>
                  </tr>
                </tbody>
                <!---->
              </table>
              <!---->
            </div>
            <!---->
          </div>
        </div>
      </div>
      <footer class="card-footer"><a class="card-footer-item">
          編集
        </a> <a class="card-footer-item">
          インポート
        </a> <a class="card-footer-item">
          全件表示
        </a></footer>
    </div>
  </transition-stub>
</div>
`;

exports[`/components/shared/ScoreBoard.vue fetch() sets { locale: en } areaName 1`] = `
Array [
  Object {
    "clearLamp": 7,
    "exScore": 414,
    "isArea": true,
    "score": 1000000,
    "userId": "0",
    "userName": "World Top",
  },
  Object {
    "clearLamp": 6,
    "exScore": 409,
    "isArea": true,
    "score": 999950,
    "userId": "13",
    "userName": "Tokyo Top",
  },
  Object {
    "clearLamp": 6,
    "exScore": 409,
    "score": 999950,
    "userId": "1user",
    "userName": "Public User",
  },
  Object {
    "clearLamp": 6,
    "exScore": 409,
    "score": 999930,
    "userId": "private_user",
    "userName": "Private User",
  },
]
`;

exports[`/components/shared/ScoreBoard.vue fetch() sets { locale: ja } areaName 1`] = `
Array [
  Object {
    "clearLamp": 7,
    "exScore": 414,
    "isArea": true,
    "score": 1000000,
    "userId": "0",
    "userName": "全国トップ",
  },
  Object {
    "clearLamp": 6,
    "exScore": 409,
    "isArea": true,
    "score": 999950,
    "userId": "13",
    "userName": "東京都トップ",
  },
  Object {
    "clearLamp": 6,
    "exScore": 409,
    "score": 999950,
    "userId": "1user",
    "userName": "Public User",
  },
  Object {
    "clearLamp": 6,
    "exScore": 409,
    "score": 999930,
    "userId": "private_user",
    "userName": "Private User",
  },
]
`;
